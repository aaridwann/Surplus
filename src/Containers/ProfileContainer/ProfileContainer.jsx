import React from 'react';
import { Text, View } from 'react-native';
import { scaleSize, scaleWidth } from '../../Utils/Scale/Scale';
import { useSelector } from 'react-redux';
import { SimpleLineIcons } from '@expo/vector-icons';
import * as Animatable from 'react-native-animatable';
import styles from './ProfileContainer.style';

const _renderGreeting = ({ name }) => (
	<Animatable.View animation={'slideInDown'} style={styles.greeting}>
		<Text style={styles.hello}>Hello </Text>
		<Text style={styles.name}>{name}</Text>
		<SimpleLineIcons
			style={{ margin: scaleWidth(5) }}
			name="diamond"
			size={scaleSize(18)}
			color="white"
		/>
	</Animatable.View>
);

function ProfileContainer() {
	const { auth } = useSelector((state) => state);
	const { username } = auth;

	return (
		<View style={styles.container}>
			<_renderGreeting name={username} />
			<Animatable.Text
				delay={10}
				animation={'slideInRight'}
				style={styles.title}
			>
				Profile Container
			</Animatable.Text>

			<Animatable.Text
				delay={10}
				animation={'slideInLeft'}
				style={styles.maintenance}
			>
				Is Under Maintenance
			</Animatable.Text>
		</View>
	);
}

export default ProfileContainer;
