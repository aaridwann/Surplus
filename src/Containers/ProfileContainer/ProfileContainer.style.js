import { StyleSheet } from 'react-native';
import { scaleFont } from '../../Utils/Scale/Scale';
import Constants from '../../Utils/Constants/Constants';

const styles = StyleSheet.create({
	webView: {
		flex: 1,
		zIndex: -1,
		top: 0,
		width: '100%',
	},
	container: {
		width: '100%',
		flex: 1,
		backgroundColor: Constants.Colors.LightGreenSurplus,
		justifyContent: 'center',
		alignItems: 'center',
	},
	title: {
		fontSize: scaleFont(30),
		color: 'white',
		fontWeight: '700',
	},
	greeting: {
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'center',
	},
	hello: {
		fontSize: scaleFont(25),
		color: 'white',
	},
	name: {
		fontSize: scaleFont(40),
		color: 'white',
		fontWeight: '600',
	},
	maintenance: {
		fontSize: scaleFont(15),
		color: 'white',
		textTransform: 'lowercase',
	},
});

export default styles;
