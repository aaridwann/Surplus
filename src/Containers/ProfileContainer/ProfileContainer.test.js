import React from 'react';
import ProfileContainer from './ProfileContainer';
import { render } from '@testing-library/react-native';

jest.mock('react-redux', () => ({
	useSelector: jest.fn().mockReturnValue({ auth: { username: '' } }),
}));

describe('Profile Container', () => {
	it('Snapshot', () => {
		expect(render(<ProfileContainer />)).toMatchSnapshot();
	});
});
