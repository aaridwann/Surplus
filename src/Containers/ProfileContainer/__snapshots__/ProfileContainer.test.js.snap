// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Container Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#50b8a8",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "transform": Array [
          Object {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 50,
        }
      }
    >
      Hello 
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 80,
          "fontWeight": "600",
        }
      }
    />
  </View>
  <Text
    collapsable={false}
    forwardedRef={[Function]}
    style={
      Object {
        "color": "white",
        "fontSize": 60,
        "fontWeight": "700",
        "transform": Array [
          Object {
            "translateX": 100,
          },
        ],
      }
    }
  >
    Profile Container
  </Text>
  <Text
    collapsable={false}
    forwardedRef={[Function]}
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "textTransform": "lowercase",
        "transform": Array [
          Object {
            "translateX": -100,
          },
        ],
      }
    }
  >
    Is Under Maintenance
  </Text>
</View>
`;
